File: finding-seams/csharp/FindingSeams.Tests/RoomAvailabilityTest.cs
using FindingSeams.RoomAvailability;
using Xunit;

namespace FindingSeams.Tests;

public class RoomAvailabilityTest
{
    private static readonly TimeSlot ExampleTimeslot = new("2026-01-11T10:00 CET", 60);

    [Fact(Skip = "Exercise: finding seams — enable when working on room availability")]
    public void Assigns_room()
    {
        RoomRegistries.Instance.Register(new Room("room-1", "Ekin Kammi"));
        var roomService = new RoomService();

        var result = roomService.PlaceSessionInRoom(1, "room-1", ExampleTimeslot);

        Assert.Equal("ACCEPTED", result.Status);
        Assert.Equal(1, result.SessionId);
        Assert.Equal("room-1", result.RoomId);
        Assert.Equal(ExampleTimeslot, result.TimeSlot);
    }

    [Fact(Skip = "Exercise: inject a shape with method is_available that you can control")]
    public void Fails_to_assign_room()
    {
        var roomService = new RoomService();
        var conflictingSlot = new TimeSlot("2026-01-11T09:30 CET", 60);

        var result = roomService.PlaceSessionInRoom(1, "room-1", conflictingSlot);

        Assert.Equal("REJECTED", result.Status);
        Assert.Equal(1, result.SessionId);
        Assert.Equal("room-1", result.RoomId);
        Assert.Equal("Room not available", result.Reason);
        Assert.Equal(conflictingSlot, result.TimeSlot);
    }
}

File: finding-seams/kotlin/src/test/kotlin/findingseams/RoomAvailabilityTest.kt
package findingseams

import findingseams.roomavailability.PlaceSessionResult
import findingseams.roomavailability.Room
import findingseams.roomavailability.RoomService
import findingseams.roomavailability.TimeSlot
import findingseams.roomavailability.roomRegistry
import org.junit.jupiter.api.Disabled
import org.junit.jupiter.api.Test
import kotlin.test.assertEquals
import kotlin.test.assertIs

@Disabled("Exercise: finding seams — enable when working on room availability")
class RoomAvailabilityTest {
    private val exampleTimeslot = TimeSlot("2026-01-11T10:00 CET", 60)

    @Test
    fun `assigns room`() {
        roomRegistry.register(Room("room-1", "Ekin Kammi"))
        val roomService = RoomService()

        val result = roomService.placeSessionInRoom(1, "room-1", exampleTimeslot)

        assertIs<PlaceSessionResult.Accepted>(result)
        assertEquals(1, result.sessionId)
        assertEquals("room-1", result.roomId)
        assertEquals("ACCEPTED", result.status)
        assertEquals(exampleTimeslot, result.timeSlot)
    }

    @Disabled("Exercise: inject a shape with method isAvailable that you can control")
    @Test
    fun `fails to assign room`() {
        val roomService = RoomService()
        val conflictingSlot = TimeSlot("2026-01-11T09:30 CET", 60)

        val result = roomService.placeSessionInRoom(1, "room-1", conflictingSlot)

        assertIs<PlaceSessionResult.Rejected>(result)
        assertEquals(1, result.sessionId)
        assertEquals("room-1", result.roomId)
        assertEquals("REJECTED", result.status)
        assertEquals("Room not available", result.reason)
        assertEquals(conflictingSlot, result.timeSlot)
    }
}
